templates/base.html

```diff
@@ -47,7 +47,7 @@
 <body class="{% block body_class %}{% endblock %}">
 {% wagtailuserbar %}
 <a href="#maincontent" class="skip-to-content-link sr-only sr-only-focusable" data-link="Skip to main content">Skip to main content</a>
-{% alerts page=page %}
+{% block alerts %}{% alerts page=page %}{% endblock %}
```

Comment:
Instead of this, I wonder if it would be better to have the following template structure?

1. `base.html` (can be subclassed for anything, regardless of whether it's a page)
2. `pages/base.html` (extends `base.html` to add page specific things (like the alerts)
3. `pages/specific_page_type.html` (extends `pages/base.html` to renders any content unique for this page type)

You will soon have all of the custom search views that will no doubt also want to extend `base.html`, so that template would do well to assume less about how it's being used.

Reply:
Wagtail pro tip: You can use the [`{% wagtail_site %}`](https://docs.wagtail.io/en/stable/topics/writing_templates.html#multi-site-support) tag to access the current `site` in the `base.html`, which you don't need a page for.

This is generally a lot more efficient, because the tag will cache the appropriate `Site` on the current `request` object, where it will be reused by things like the `pageurl` tag and retrieving site-specific settings. So, although it isn't 'free' to use - it's likely that query is already being triggered by other code, so you might as well profit. `Page.get_site()` always triggers at least two database/cache backend queries.

Reply:
Thanks Andy! That's a good tip. I've made the change to `base.html` and removed `Record.get_site`.

The template restructuing makes sense. It might be worth opening the discussion up to a wider audience, particularly frontend. Worth discussing in today's dev meeting?

Reply:
> Worth discussing in today's dev meeting?

I have no time scheduled for this week unfortunately (unsure what is happening contractually), but would be good if you could mention it :)